pkg/logger/logger.go

```diff
@@ -60,19 +117,29 @@ func ZapLogger(cfDebug bool, cfLogLevel int) logr.Logger {
 		StacktraceKey:  "stacktrace",
 		LineEnding:     zapcore.DefaultLineEnding,
 		EncodeTime:     customTimeEncoder,
-		EncodeLevel:    zapcore.CapitalColorLevelEncoder,
+		EncodeLevel:    customLevelEncoder, // Use custom level encoder
 		EncodeCaller:   customCallerEncoder,
 		EncodeDuration: zapcore.SecondsDurationEncoder,
 		EncodeName:     zapcore.FullNameEncoder,
 	}
-	core := zapcore.NewCore(
+
+	baseCore := zapcore.NewCore(
 		zapcore.NewConsoleEncoder(encoderConf),
 		zapcore.AddSync(zapcore.Lock(os.Stdout)),
 		zapcore.Level(-1*logLevel),
 	)
-	zapLogger := zap.New(core, zap.AddCaller(), zap.AddCallerSkip(0))
 
-	defer zapLogger.Sync()
+	filteredCore := &FilteredCore{
+		Core: baseCore,
+		// Filter recorder/recorder.go log
+		filter: filterRecorderLogs,
+	}
+
+	zapLogger := zap.New(filteredCore, zap.AddCaller(), zap.AddCallerSkip(0))
+
+	defer func(zapLogger *zap.Logger) {
+		_ = zapLogger.Sync()
+	}(zapLogger)
 
```

Comment:
the same as "defer zapLogger.Sync()"?

Reply:
It is to avoid IDE warning non error handling